core/headerchain.go

```diff
@@ -591,9 +609,9 @@ func (hc *HeaderChain) SetHead(head uint64, updateFn UpdateHeadBlocksCallback, d
 		markerBatch := hc.chainDb.NewBatch()
 		if updateFn != nil {
 			newHead, force := updateFn(markerBatch, parent)
-			if force && newHead < head {
-				log.Warn("Force rewinding till ancient limit", "head", newHead)
-				head = newHead
+			if force && ((headTime > 0 && newHead.Time < headTime) || (headTime == 0 && newHead.Number.Uint64() < headBlock)) {
```

Comment:
Isn't this just the same condition as `done(newHead)`? It passes the tests when I replace it with that

Reply:
Almost. Done stops when newHead.Number <= headBlock, this one only cares if it's <, if it's ==, then no more rewind is done.